datastream-utils/src/main/java/com/linkedin/datastream/common/FutureUtils.java

```diff
@@ -0,0 +1,43 @@
+/**
+ *  Copyright 2019 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.common;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+
+public class FutureUtils {
+
+  private static final ScheduledExecutorService SCHEDULED_EXECUTOR_SERVICE = Executors.newScheduledThreadPool(100);
+
+  /**
+   * This function waits till timeout for Future to complete, otherwise, it cancels it.
+   * @param futureToAwait Future to complete/stop
+   * @param timeout time to wait before aborting future
+   * @return CompletableFuture for this operation
+   */
+  public static CompletableFuture<?> awaitFutureOrCancel(Future<?> futureToAwait, Duration timeout) {
```

Comment:
This comment is purely about naming.

- I think it'd be better to rename `futureToAwait` to just `future`
- If you decide to go with an implementation that propagates back the future's result, I'd consider giving the method a different name, one that reflects that it is meant to get the future's result (if possible) after a specific duration (e.g. `getIfDoneOrCancel`?)
- `timeout` is a misleading name for that param cause it implies we may check/finish sooner. We need a name that convey that this is how much we'll wait (should we call it `after`?)